platform/darwin/src/MGLBackgroundStyleLayer.mm

```diff
@@ -62,13 +62,13 @@ - (void)addToMapView:(MGLMapView *)mapView belowLayer:(MGLStyleLayer *)otherLaye
 
 - (void)removeFromMapView:(MGLMapView *)mapView
 {
-    _pendingLayer = nullptr;
-    self.rawLayer = nullptr;
-
     auto removedLayer = mapView.mbglMap->removeLayer(self.identifier.UTF8String);
     if (!removedLayer) {
         return;
     }
+    auto originalLayer = std::make_unique<mbgl::style::BackgroundLayer>(self.identifier.UTF8String);
+    _pendingLayer = std::move(originalLayer);
+    self.rawLayer = _pendingLayer.get();
```

Comment:
If I understand correctly, these new lines are only relevant to the case where we bail below due to removing a layer of a different type than this class, correct? Otherwise, we’ll end up making the removed layer the pending layer.